sechub-wrapper-owasp-zap/src/test/java/com/mercedesbenz/sechub/zapwrapper/config/ZapWrapperDataSectionFileProviderTest.java

```diff
@@ -0,0 +1,366 @@
+                "webScan":{"url":"https://localhost:8443","api":{"type":"openApi","use":["no-existing-reference"]}}}""";
+        SecHubScanConfiguration sechubScanConfiguration = SecHubScanConfiguration.createFromJSON(sechubScanConfigJSON);
+
+        /* execute */
+        List<File> result = providerToTest.fetchApiDefinitionFiles("test/path", sechubScanConfiguration);
+
+        /* test */
+        assertTrue(result.isEmpty());
+    }
+
+    @Test
+    void client_cert_data_section_name_differs_from_use_part_inside_openapi_definition_results_in_null_as_cert_file() {
+        /* prepare */
+        String sechubScanConfigJSON = """
+                {"apiVersion":"1.0","data":{"sources":[{"name":"client-cert-file-reference","fileSystem":{"files":["clientCert.p12"]}}]},
+                "webScan":{"url":"https://localhost:8443","clientCertificate":{"password":"secret","use":["no-existing-reference"]}}}""";
+        SecHubScanConfiguration sechubScanConfiguration = SecHubScanConfiguration.createFromJSON(sechubScanConfigJSON);
+
+        /* execute */
+        File result = providerToTest.fetchClientCertificateFile("test/path", sechubScanConfiguration);
+
+        /* test */
+        assertNull(result);
+    }
+
+    @Test
+    void open_api_valid_sechub_scan_config_with_openapi_definition_file_results_in_list_with_one_file() {
+        /* prepare */
+        String sechubScanConfigJSON = """
+                {"apiVersion":"1.0","data":{"sources":[{"name":"open-api-file-reference","fileSystem":{"files":["openapi3.json"]}}]},
+                "webScan":{"url":"https://localhost:8443","api":{"type":"openApi","use":["open-api-file-reference"]}}}""";
+        SecHubScanConfiguration sechubScanConfiguration = SecHubScanConfiguration.createFromJSON(sechubScanConfigJSON);
+
+        /* execute */
+        List<File> result = providerToTest.fetchApiDefinitionFiles("test/path", sechubScanConfiguration);
+
+        /* test */
+        assertEquals(1, result.size());
+    }
+
+    @Test
+    void client_cert_valid_sechub_scan_config_with_openapi_definition_file_results_in_client_cert_file() {
+        /* prepare */
+        String sechubScanConfigJSON = """
+                {"apiVersion":"1.0","data":{"sources":[{"name":"client-cert-file-reference","fileSystem":{"files":["clientCert.p12"]}}]},
+                "webScan":{"url":"https://localhost:8443","clientCertificate":{"password":"secret","use":["client-cert-file-reference"]}}}""";
+        SecHubScanConfiguration sechubScanConfiguration = SecHubScanConfiguration.createFromJSON(sechubScanConfigJSON);
+
+        /* execute */
+        File result = providerToTest.fetchClientCertificateFile("test/path", sechubScanConfiguration);
+
+        /* test */
+        assertEquals("test/path/clientCert.p12", result.toString());
+    }
+
+    @ParameterizedTest
+    @MethodSource("multipleFilesTestNamedArguments")
+    void open_api_mutliple_files_result_in_correct_list_of_files(String sechubScanConfigJSON) {
```

Comment:
I am not sure, whether I understood the purpose of the tests:

```suggestion
    void multiple_open_api_files_result_in_multiple_open_api_files_available(String sechubScanConfigJSON) {
```